pkg/ctrl/watch/pod.go

```diff
@@ -98,24 +99,31 @@ func shouldRequeue(pod *corev1.Pod) bool {
 		return false
 	}
 
+	// ignore if pod status is not running
+	if pod.Status.Phase != corev1.PodRunning || len(pod.Status.ContainerStatuses) < 2 {
+		log.Info("Pod status is not running or containerStatus less than 2.", "name", pod.Name, "namespace", pod.Namespace)
+		return false
+	}
+
 	// reconcile if all app containers exit 0 and fuse container not exit
 	appExited := true
-	fuseExited := false
+	fuseRunning := false
 	for _, containerStatus := range pod.Status.ContainerStatuses {
 		if containerStatus.Name != common.FuseContainerName {
-			if containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode == 0 {
+			log.V(2).Info("container status", "status", containerStatus)
+			if containerStatus.State.Terminated != nil {
 				log.Info("fluid app exited", "pod", pod.Name, "container", containerStatus.Name, "namespace", pod.Namespace)
 			} else {
 				// container not exist
 				appExited = false
```

Comment:
If app Container is not exited or fuseContainer is not running,  how about `break` the loop immediately?